From 8dcc75f524b1de24b2ab2829e3c9b97ac195903e Mon Sep 17 00:00:00 2001
From: Pavel Nedev <pnedev@mm-sol.com>
Date: Fri, 28 Jan 2011 13:09:11 +0200
Subject: OMAP4/DSS: Removed unnecessary clock toggling in Overlay

When setting overlay to manager in omap_dss_set_manager overlay function the DSS
interface and functional clocks were unnecessary toggled.
Delay before setting overlay manager also removed as the described issue is no longer observed.
In case main_clk is disabled return value is changed to -EIO.

Change-Id: Ib409d27bbb01cc6aac167f20042e22df77155ce7
Signed-off-by: Pavel Nedev <pnedev@mm-sol.com>
---
 drivers/video/omap2/dss/overlay.c |   15 ++++-----------
 1 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/video/omap2/dss/overlay.c b/drivers/video/omap2/dss/overlay.c
index d5ccd14..6810ba5 100644
--- a/drivers/video/omap2/dss/overlay.c
+++ b/drivers/video/omap2/dss/overlay.c
@@ -635,19 +635,12 @@ static int omap_dss_set_manager(struct omap_overlay *ovl,
 	ovl->manager = mgr;
 
 	/* do not set channel out if DSS is off */
-	if (!dss_get_mainclk_state())
-		return 0;
+	if (!dss_get_mainclk_state()) {
+		DSSERR("DSS clock not active, setting ovl manager failed\n");
+		return -EIO;
+	}
 
-	dss_clk_enable(DSS_CLK_ICK | DSS_CLK_FCK1);
-	/* XXX: on manual update display, in auto update mode, a bug happens
-	 * here. When an overlay is first enabled on LCD, then it's disabled,
-	 * and the manager is changed to TV, we sometimes get SYNC_LOST_DIGIT
-	 * errors. Waiting before changing the channel_out fixes it. I'm
-	 * guessing that the overlay is still somehow being used for the LCD,
-	 * but I don't understand how or why. */
-	msleep(40);
 	dispc_set_channel_out(ovl->id, mgr->id);
-	dss_clk_disable(DSS_CLK_ICK | DSS_CLK_FCK1);
 
 	return 0;
 }
-- 
1.6.0.4

